### DISCUSSION TITLE
SVG Image press effect

### DISCUSSION THREAD

**Discussion by WittObk** [ElinamLLC/SharpVectors#218] [Q&A]
I use some svg images for buttons and want to have a press effect for it ... is it possible to change the background color of the svg image? I use the FileSvgReader to load the image ...

**Comment by paulushub** [comment]
For the **background**, since WPF image control does not have background property, you can do one of the following:
1. Put the Image control in Border control and change the `Border.Background` color. Set the effect in a XAML trigger or use mouse events.
2. Create a background layer, after the conversion as shown in the code below (currentCurrent is the result from FileSvgReader). Give a name to the background brush, and use either trigger or events to change it.
```cs
private DrawingGroup WrapDrawing(DrawingGroup currentDrawing)
{
    var bounds = currentDrawing.Bounds;
    var backgroundLayer = new DrawingGroup();
    var backgroundBrush = new SolidColorBrush(Colors.Transparent);
    backgroundBrush.SetValue(NameProperty, "backbrush");               // <===== Give it a name to help you control it.
    var backgroundDrawing = new GeometryDrawing(backgroundBrush,
        new Pen(Brushes.Red, 1), new RectangleGeometry(bounds));       // <===== Add a border color with a pen if you need it
    backgroundLayer.Children.Add(backgroundDrawing);

    var combinedDrawing = new DrawingGroup();
    combinedDrawing.Children.Add(backgroundLayer);
    combinedDrawing.Children.Add(currentDrawing);

    return combinedDrawing;
}
```
